src/hivpy/sexual_behaviour.py

```diff
@@ -70,86 +98,234 @@ def __init__(self, **kwargs):
         self.ltp_balance_factor = {SexType.Male: 1, SexType.Female: 1}
         self.new_ltp_age_bins = [35, 45, 55]
 
+        # sex workers parameters
+        self.base_start_sw = self.sb_data.base_start_sex_work.sample()
+        self.base_stop_sw = self.sb_data.base_stop_sex_work.sample()
+        self.risk_sex_worker_age = self.sb_data.risk_sex_worker_age
+        self.prob_init_sex_work_age = self.sb_data.prob_init_sex_work_age
+        self.sw_age_bins = [20, 25, 35]
+        self.sex_worker_program = True if rng.random() < 0.2 else False
+        self.sw_program_start_date = datetime.date(2015, 1, 1)
+        self.prob_sw_program_effect = 1.0
+        self.rate_engage_sw_program = 0.1
+        self.rate_disengage_sw_program = 0.025
+        self.sw_program_cost = 0.01  # placeholder
+        self.prob_high_sex_risk = self.sb_data.probability_high_sexual_risk.sample()
+        self.incr_rate_sw_high_sex_risk = self.sb_data.incr_rate_sw_high_sex_risk
+        self.prob_sw_program_effective = self.sb_data.prob_sw_program_effect.sample()
+
     def age_index(self, age):
         return np.minimum((age.astype(int)-self.risk_min_age) //
                           self.risk_age_grouping, self.risk_categories)
 
+    def init_sex_behaviour(self, population: Population):
+        population.init_variable(col.NUM_PARTNERS, 0, data_type=pd.Int32Dtype)
+        population.init_variable(col.RISK, 1)
+        population.init_variable(col.LONG_TERM_PARTNER, False)
+        population.init_variable(col.LTP_AGE_GROUP, 0)
+        population.init_variable(col.LTP_LONGEVITY, 0)
+        population.init_variable(col.SEX_MIX_AGE_GROUP, 0)
+        population.init_variable(col.STP_AGE_GROUPS, np.array([[0]]*population.size))
+        population.init_variable(col.RISK_LTP, 1)
+        population.init_variable(col.ADC, False)
+        population.init_variable(col.LIFE_SEX_RISK, 1)
+        population.init_variable(col.SEX_WORKER, False)
+        population.init_variable(col.SW_TEST_6MONTHLY, False)
+        population.init_variable(col.SW_PROGRAM_VISIT, False)
+        population.init_variable(col.AGE_STOP_SEX_WORK, None)
+        population.init_variable(col.SW_AGE_GROUP, 0)
+        population.init_variable(col.DATE_STOP_SW, None)
+        population.init_variable(col.SEX_BEHAVIOUR_CLASS, 0)
+        self.init_risk_factors(population)
+        self.init_sex_worker_status(population)
+        self.init_sex_behaviour_groups(population)
+        self.num_short_term_partners(population)
+
     def update_sex_behaviour(self, population: Population):
+        self.update_risk(population)
+        self.update_sex_groups(population)
         self.num_short_term_partners(population)
         self.assign_stp_ages(population)
-        self.update_sex_groups(population)
-        self.update_rred(population)
         self.update_long_term_partners(population)
+        self.update_sex_worker_status(population)
+        # self.update_sex_behaviour_class(population)
```

Comment:
So having looked at this again I think that actually this should be uncommented and update sex worker status and sex behaviour class will be moved to the top of update_sex_behaviour. The reason for this is:
1. Sex worker / behaviour class status should affect the number of partners for _this_ time step
2. I had previously assumed that sex behaviour class only changes because of changes to the sex worker status, but it can also occur because of age (for women), so I think it's clearer to move it outside of the update_sex_worker_status method